awsproviderlint/passes/AWSAT003/AWSAT003.go

```diff
@@ -0,0 +1,65 @@
+// Package AWSAT003 defines an Analyzer that checks for
+// hardcoded regions and availability zones
+package AWSAT003
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"regexp"
+
+	"github.com/bflad/tfproviderlint/passes/commentignore"
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+)
+
+const Doc = `check for hardcoded regions and AZs
+
+The AWSAT003 analyzer reports hardcoded regions and AZs. Testing
+in non-commercial partitions with hardcoded regions and AZs will cause the 
+tests to fail. 
+`
+
+const analyzerName = "AWSAT003"
+
+var Analyzer = &analysis.Analyzer{
+	Name: analyzerName,
+	Doc:  Doc,
+	Requires: []*analysis.Analyzer{
+		commentignore.Analyzer,
+		inspect.Analyzer,
+	},
+	Run: run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	ignorer := pass.ResultOf[commentignore.Analyzer].(*commentignore.Ignorer)
+	inspect := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
+
+	nodeFilter := []ast.Node{
+		(*ast.BasicLit)(nil),
+	}
+
+	majorRegions := "ap|eu|sa|us"
+	directions := "central|east|northeast|south|southeast|west"
+	re := regexp.MustCompile(fmt.Sprintf("(%q)-(%q)", majorRegions, directions) + `-\d`)
+	inspect.Preorder(nodeFilter, func(n ast.Node) {
+		x := n.(*ast.BasicLit)
+
+		if ignorer.ShouldIgnore(analyzerName, x) {
+			return
+		}
+
+		if x.Kind != token.STRING {
+			return
+		}
+
+		if !re.MatchString(x.Value) {
+			return
+		}
+
+		pass.Reportf(x.ValuePos, "%s: regions and AZs should not be hardcoded", analyzerName)
```

Comment:
It might be nice to give developers a hint about how to fix these in the message as well, e.g.

```suggestion
		pass.Reportf(x.ValuePos, "%s: regions and AZs should not be hardcoded, use the aws_availability_zones and aws_region data sources instead", analyzerName)
```